.github/workflows/pypi-package.yml

```diff
@@ -0,0 +1,44 @@
+name: Build & maybe upload PyPI package
+
+on:
+  push:
+  pull_request:
+  release:
+    types:
+      - published
+  workflow_dispatch:
+
+permissions:
+  contents: read
+
+jobs:
+  # Always build & lint package.
+  build-package:
+    name: Build & verify package
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - uses: hynek/build-and-inspect-python-package@v1
+
+  # Upload to real PyPI on GitHub Releases.
+  release-pypi:
+    name: Publish to PyPI
+    environment: release-pypi
+    if: github.repository_owner == 'python' && github.event.action == 'published'
```

Comment:
I would add a short comment to clarify when this step is expected to be executed.

Reply:
Added!